src/platform/macos/misc.mm

```diff
@@ -53,13 +53,11 @@
     // Xcode 12.2 and later, these functions are not weakly linked and will never
     // be null, and therefore generate this warning. Since we are weakly linking
     // when compiling with earlier Xcode versions, the check for null is
-    // necessary and so we ignore the warning.
+    // necessary, and so we ignore the warning.
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wunguarded-availability-new"
 #pragma clang diagnostic ignored "-Wtautological-pointer-compare"
-    if ([[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:((NSOperatingSystemVersion) { 10, 15, 0 })] &&
-        // Double check that these weakly-linked symbols have been loaded:
-        CGPreflightScreenCaptureAccess != nullptr && CGRequestScreenCaptureAccess != nullptr &&
+    if ([[NSProcessInfo processInfo] isOperatingSystemAtLeastVersion:((NSOperatingSystemVersion) { 10, 15, 0 })] != 0 && YES &&
```

Comment:
This change should probably be reverted or modified. `&& YES` does nothing (`YES` is `true`).

Reply:
For reference, these were the warnings.

![image](https://github.com/LizardByte/Sunshine/assets/42013603/223f3d66-a0f1-4bb1-b627-402dda41797e)

But yea, what it corrected it to doesn't seem logical.

I'm also thinking that the always true warnings are false positives since my VM is a higher version than macOS 11.